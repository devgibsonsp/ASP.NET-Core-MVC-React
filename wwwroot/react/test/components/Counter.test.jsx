import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Counter from 'ui/components/Counter';

describe('Match <Counter/> Snapshot', () => {
   it('Matches the previous snapshot of the component', () => {
     const component = shallow((
       <Counter />
     ));
     expect(shallowToJson(component)).toMatchSnapshot();
   });
 });

describe('Test <Counter/>', () => {
   
     it('renders the counter component successfully', () => {
        const wrapper = shallow(<Counter/>);
        expect(wrapper.exists()).toEqual(true);
     });

     it('increments the counter when clicked', () => {
        const wrapper = shallow(<Counter/>);
        const instance = wrapper.instance();
        expect(instance.state.currentCount).toEqual(0);
        instance.incrementCounter();
        expect(instance.state.currentCount).toEqual(1);

     });
  });

