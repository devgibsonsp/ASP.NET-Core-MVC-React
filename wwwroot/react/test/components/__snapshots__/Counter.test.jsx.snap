// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match <Counter/> Snapshot Matches the previous snapshot of the component 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <p>
    This is a simple example of a React component.
  </p>
      
  <p>
    Current count: 
    <strong>
      0
    </strong>
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
